src/test/compile-fail-fulldeps/proc-macro/shadow.rs

```diff
@@ -9,13 +9,12 @@
 // except according to those terms.
 
 // aux-build:derive-a.rs
-// aux-build:derive-a-2.rs
 
 #![feature(proc_macro)]
 
 #[macro_use]
 extern crate derive_a;
 #[macro_use]
-extern crate derive_a_2; //~ ERROR: cannot shadow existing derive mode `A`
+extern crate derive_a; //~ ERROR `derive_a` has already been defined
```

Comment:
Yeah, it used to test that a `#[macro_use]`-imported custom derive cannot shadow an existing `#[macro_use]`-imported custom derive.

This PR changes `#[macro_use]`-imported custom derives to have the same scope and shadowing restrictions as other `#[macro_use]`-imported macros, so it is no longer an error for an unexpanded `#[macro_use]` to import a shadowing custom derive (just as it is not an error ordinary unexpanded `#[macro_use]`s to shadow).